src/server/redis_connection.cc

```diff
@@ -74,6 +75,17 @@ void Connection::Close() {
 
 void Connection::Detach() { owner_->DetachConnection(this); }
 
+bool Connection::IsBlockingMode() const {
+  if (current_cmd == nullptr) return false;
+  if (dynamic_cast<BlockingCommander *>(current_cmd.get()) != nullptr) return true;
+
+  auto args = current_cmd->GetArgs();
+  if (args.empty()) return false;
+
+  std::string cmd_name = util::ToLower(args.front());
+  return cmd_name == "xread";
```

Comment:
Seems it can cause data race since args are used in multiple threads. And the check `cmd_name == "xread"` is really a disaster of maintainability (we can easily forget while facing new blocking commands).

So I think a better solution is add an new attribute like `blocking` to these commands and check it here.

Reply:
Gotcha, thank you!